internal/controller/certificates/policies/policies.go

```diff
@@ -104,6 +104,7 @@ func NewReadinessPolicyChain(c clock.Clock) Chain {
 // correctness of metadata and output formats of Certificate's Secrets.
 func NewSecretPostIssuancePolicyChain(ownerRefEnabled bool, fieldManager string) Chain {
 	return Chain{
+		SecretCertificateNameAnnotationsMismatch,                             // Make sure the Secret's CertificateName annotation matches the Certificate's name
```

Comment:
I wonder if we should focus on what/why we are checking this, and not how it's implemented? I know it's optional for a certificate to technically own/control the secret, based on resource owner references, but I still think it could make sense to use a wording like e.g. `SecretOwnershipMismatch` here?

Reply:
The primary reason to use `SecretCertificateNameAnnotationsMismatch` is to re-use existing implementation. On the wording, I feel, `SecretOwnershipMismatch` would lead to ambiguity in for annotation vs `metadata.ownerReferences` check.

Reply:
@TrilokGeer I think it would be better to check the annotation outside of the policy chain.
This solution depends on the ordering of the elements in the chain (which was not that important before).
The chain normally acts as an OR operation chaining together different conditions that can trigger a post-issuance step, but in this PR we are introducing an additional condition (the Annotation must not mismatch AND the other elements in the chain must return true).

I propose we move the check to eg line 67 in `pkg/controller/certificates/issuing/secret_manager.go`.

Reply:
Thanks for the review @inteon , prioritized to take a look at this this week.